### ISSUE TITLE
[MU4 Issue] [Accessibility] Text editing events are not announced by screen readers

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#10234]
**Describe the bug**
When entering lyrics or stave text, or when typing in a text frame, screen readers should announce text characters as they are typed, selected, deselected and navigated with the cursor keys. At the moment, the screen reader is silent during this process.

![image](https://user-images.githubusercontent.com/11011881/148830897-debe0bea-b75e-4eda-af83-651c06aef91a.png)

**To Reproduce**
1. Start the screen reader (Narrator, NVDA, VoiceOver, Orca, etc.).
2. Click on a note or rest in any score.
3. Press Ctrl+T (macOS: Cmd+T) to enter stave text.
4. Start typing text characters and listen to the screen reader.
5. Use the arrow keys to navigate through the typed character. Keep listening to the screen reader.
6. Hold Shift and continue pressing arrow keys to select and deselect typed characters. Keep listening to the screen reader.
7. Repeat steps 4 to 6 in your platform's native text editor (e.g. Notepad, TextEdit) and compare the screen reader experience.

**Expected behavior**
The screen reader should behave the same in MuseScore during text editing as it does in the platform's native text editor. It should announce text characters as they are typed, navigated, selected and deselected.

**Desktop (please complete the following information):**
 - OS: Windows, macOS, Linux

**Additional context**
In MuseScore 3, this was achieved by [sending a message to the screen reader](https://github.com/musescore/MuseScore/blob/3.x/mscore/scoreaccessibility.cpp#L93-L103) that contained the [exact text characters that should be spoken](https://github.com/musescore/MuseScore/blob/2513676e512d29d554cb6c4d37d3efaf53ea2c5b/libmscore/textbase.cpp#L555-L670), but this was more of a hack than an actual solution.

In MuseScore 4 it should be achieved by subclassing [QAccessibleTextInterface](https://doc.qt.io/qt-5/qaccessibletextinterface.html) similarly to how [we already subclass QAccessibleInterface and QAccessibleValueInterface](https://github.com/musescore/MuseScore/blob/f30d808e5dd84fc732291ab3b222dfa2c55095ae/src/framework/accessibility/internal/accessibleiteminterface.h#L33). However, I'm not sure how we would integrate this QAccessibleTextInterface subclass into the accessibility tree given that it would be used by some elements but not others.
